File: app/src/main/java/app/we/go/emojidraw/arch/di/ApplicationComponent.java
package app.we.go.emojidraw.arch.di;

import javax.inject.Singleton;

import app.we.go.emojidraw.ThisApplication;
import app.we.go.emojidraw.arch.di.features.practice.PracticeModule;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Singleton
@Component(modules = {
        AndroidInjectionModule.class,
        ApplicationModule.class,
        ServiceModule.class,
        PracticeModule.class})
public interface ApplicationComponent {

    void inject(ThisApplication thisApplication);
}

File: app/src/main/java/app/we/go/emojidraw/arch/di/ApplicationComponent.kt
package app.we.go.emojidraw.arch.di

import app.we.go.emojidraw.arch.di.features.practice.PracticeComponent
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [(ApplicationModule::class), (ServiceModule::class)])
interface ApplicationComponent {

    fun plusPracticeComponent(): PracticeComponent

}
